edgelet/doc/devguide.md

```diff
@@ -1,163 +1,264 @@
+
+After installing `rustup`, or if you already have it installed, install the toolchain that will be used to build the daemon binaries.
+
+```sh
+rustup self update   # Update rustup itself to the latest version
+
+git clone --recurse-submodules 'https://github.com/Azure/iotedge'
+cd iotedge/edgelet/
+
+rustup update   # Install / update the toolchain used to build the daemon binaries. This is controlled by the rust-toolchain file in this directory.
+                # For the master branch, this is the latest "stable" toolchain. For release branches, this is a pinned Rust release.
 ```
 
-### Additional Dependencies
+In addition, building the daemon binaries also requires these dependencies to be installed:
+
+#### CentOS 7
+
+```sh
+yum update
+yum install \
+    cmake curl git make rpm-build \
+    gcc gcc-c++ \
+    libcurl-devel libuuid-devel openssl-devel
+```
 
-#### Linux
+#### Debian 8-10, Ubuntu 16.04, Ubuntu 18.04
 
-```bash
-sudo apt-get update
-sudo apt-get install -y git cmake build-essential curl libcurl4-openssl-dev libssl-dev uuid-dev pkg-config
+```sh
+apt-get update
+apt-get install \
+    binutils build-essential ca-certificates curl cmake debhelper dh-systemd file git make \
+    gcc g++ pkg-config \
+    libcurl4-openssl-dev libssl-dev uuid-dev
 ```
 
-### macOS
+#### macOS
+
+1. Install the dependencies using [Homebrew](https://brew.sh/) package manager
 
-1. Install necessary tools for development using [Homebrew](https://brew.sh/) package manager
-    ```bash
+    ```sh
     brew update
     brew install cmake openssl
     ```
 
-1. Set `OPENSSL_ROOT_DIR`
-    ```bash
-    export OPENSSL_ROOT_DIR=/usr/local/opt/openssl
+1. Set the `OPENSSL_DIR` and `OPENSSL_ROOT_DIR` environment variables to point to the local openssl installation.
+
+    ```sh
     export OPENSSL_DIR=/usr/local/opt/openssl
+    export OPENSSL_ROOT_DIR=/usr/local/opt/openssl
     ```
 
 #### Windows
 
+1. Install Visual Studio 2017 / 2019, or the Build Tools for Visual Studio 2017 / 2019. Ensure the components for building C / C++ are installed.
+
+1. Install `cmake` from <https://cmake.org/> or with [`choco`](https://chocolatey.org/) or [`scoop`.](https://scoop.sh/) Ensure `cmake` is in `PATH` after installation.
+
 1. Install `vcpkg`
 
-	```powershell
-	git clone https://github.com/Microsoft/vcpkg
-	cd vcpkg
-	.\bootstrap-vcpkg.bat
-	```
+    ```powershell
+    git clone https://github.com/Microsoft/vcpkg
+    cd vcpkg
+    .\bootstrap-vcpkg.bat
+    ```
+
+1. Install openssl
+
+    ```powershell
+    .\vcpkg install openssl:x64-windows
+    ```
 
-1. Install openssl binaries
+1. Set the `OPENSSL_DIR` and `OPENSSL_ROOT_DIR` environment variables to point to the local openssl installation.
 
-	```powershell
-	.\vcpkg install openssl:x64-windows
-	```
+    ```powershell
+    # $PWD is the root of the vcpkg repository
 
-1. Set `OPENSSL_ROOT_DIR`
+    $env:OPENSSL_DIR = "$PWD\installed\x64-windows"
```

Comment:
Shouldn't both of them to lead to openssl lib folder inside "$PWD\installed\x64-windows"? I remember having problems with doing as the old doc says.

Reply:
The path for `libcrypto.dll` must be `$env:OPENSSL_DIR\bin\libcrypto.dll`

See https://github.com/Azure/iotedge/blob/70f9fbe7ea18de4360351b99dfd2288c1111ee7b/edgelet/build/windows/openssl.ps1#L60-L61

Reply:
Ok, maybe it was something else.